### PR TITLE
WIP: Ability to use multiple QB instances, refactored logs format and bug fixes

### PR DESCRIPTION
**Pull request by soulfly**
**Made/Proposed changes:**
1. ability to use multiple QB instances
2. refactored logs format
3. fixed an issue with recursion when do `QB.chat.disconnect` inside `QB.chat.connect`'s callback (Node.js)
4. new callback `onKickOccupant(dialogId, initiatorUserId)` to handle a case where User2 joined chat dialog and User1 removed him from occupants. 
5. Fixed: `QB.chat.privacylist.delete` method does not work under Node.js.
6. Fixed:  methods to decline the use of active/privacy lists  (`QB.chat.privacylist.setAsDefault('', callback)` and `QB.chat.privacylist.setAsActive('', callback)`) does not work properly. Also, now you can pass `null` instead of empty string `''` to decline.
7. Fixed: `QB.chat.privacylist.getNames` method does not return anything if a user does not have any active/default lists.
8. Fixed: `QB.chat.privacylist.getNames` returns all names except active/default at `names` key under Node.js. Made it similar to browser env behaviour. 
9. Added task `npm run buildNotMinified`
10. Added Karma/Istanbul code coverage tools. To start use the following comand: `node_modules/.bin/karma start karma.conf.js`. To see a report look at  subdirectory `coverage` in the default location. **Warning!** Before run you need to rebuild sdk with `buildNotMinified` task.
11. new callbacks `onJoinOccupant(dialogId, userId)` and `onLeaveOccupant(dialogId, userId)` to handle group dialog changes. 
12. `QB.chat.muc.listOnlineUsers` method not returns an array of integers instead of array of strings. 
13. Fixed: `onMessageErrorListener` does not work under Node.js
14. Fixed: `QB.chat.privacylist.update` method does not work properly when change deny-> allow.
15. Fixed: `QB.users.resetPassword` always returns an error
16. more Chat tests

**How should this be manually tested?**
do 'npm install'
run tests

**Does the documentation needs an update?**
1. to describe new callback functionality `onKickOccupant(dialogId, initiatorUserId)`
2. to describe new callback functionality `onJoinOccupant(dialogId, userId)`
3. to describe new callback functionality `onLeaveOccupant(dialogId, userId)`

### CODE REVIEW

File: spec/QB-UsersSpec.js [hunk 1]
```diff
@@ -135,13 +135,14 @@ describe('Users API', function() {
           expect(res).not.toBeNull();
           expect(res.full_name).toBe('QuickBlox Test');
           userId = res.id;
+          console.info(userId);
```
Comment (dimaspirit) [inline spec/QB-UsersSpec.js]: remove console or this need for something?

File: spec/QB-UsersSpec.js [hunk 2]
```diff
@@ -160,17 +161,48 @@ describe('Users API', function() {
...
+    });
+
+  });
+
+  describe('Service operations: ', function(){
+
+    it('can reset password', function(done) {
+
+      QB.users.resetPassword(QBUser1.email, function(err, res){
+        console.log(err);
```
Comment (dimaspirit) [inline spec/QB-UsersSpec.js]: Do we need this console?

File: spec/QB-UsersSpec.js [hunk 3]
```diff
@@ -160,17 +161,48 @@ describe('Users API', function() {
...
+
+        done();
+      });
+
+    }, REST_REQUESTS_TIMEOUT);
+
   });
+
+  afterAll(function(done){
+    QB.destroySession(function (err, result){
```
Comment (dimaspirit) [inline spec/QB-UsersSpec.js:199]: Why doesn't check err or result?

File: spec/config.js [hunk 4]
```diff
@@ -1,37 +1,77 @@
...
-        'authSecret': 'TqQmBFbANJ6cfu4'
-    };
-
-    var QBUser1 = {
-            'id': 23286054,
-            'login': "js_jasmine11",
-            'password': "js_jasmine11",
-            'email': "js_jasmine11@quickblox.com"
+
+    var NEW_SERVER = false;
```
Comment (dimaspirit) [inline spec/config.js:5]: What is the New_server?

**Comment by dimaspirit** [comment]
Also, I remove the `src/modules/qbLocation.js`, in your pull request it returned, so need to merge or remove this file manually.

**Comment by dimaspirit** [comment]
Also, need to describe in release notes that need to update npm dependencies.

File: spec/config.js (hunk 4 above)
Comment (soulfly) [reply spec/config.js:5]: This is our new implementation with fully clustered Chat server solution, will be released in some near future.

File: spec/QB-UsersSpec.js (hunk 3 above)
Comment (soulfly) [reply spec/QB-UsersSpec.js:199]: will remove

File: spec/QB-UsersSpec.js (hunk 1 above)
Comment (soulfly) [reply spec/QB-UsersSpec.js]: will remove

File: spec/QB-UsersSpec.js (hunk 2 above)
Comment (soulfly) [reply spec/QB-UsersSpec.js]: will remove

**Pull request merged**